AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java

```diff
@@ -363,13 +376,46 @@ private void updateParticipantViews() {
     }
 
     private void showRecordingActiveNotification() {
-        if (callingContext.isRecordingActive()) {
-            setRecordingActivityNotification(getText(R.string.start_recording));
+        final Boolean isRecordingActive = callingContext.isRecordingActive();
+        final Boolean isTranscriptionActive = callingContext.isTranscriptionActive();
+
+        if (isRecordingActive) {
+            if (isTranscriptionActive) {
+                setRecordingActivityNotification(getText(R.string.start_recording_and_transcribing));
+            } else {
+                setRecordingActivityNotification(getText(R.string.start_recording));
+            }
+        } else {
+            if (isTranscriptionActive) {
+                setRecordingActivityNotification(getText(R.string.stop_recording_while_transcribing));
+            } else {
+                setRecordingActivityNotification(getText(R.string.stop_recording));
+            }
+        }
+    }
+
+    private void showTranscriptionActiveNotification() {
```

Comment:
I think we can combine these methods into one. They have the same logic to check `isRecordingActive` and `isTranscriptionActive` and display the text based on the two boolean values.

Reply:
Yea, originally I want to merge them into one. But the problem is for the case when one is true and the other is false. For this case, should we say "A has stopped. You are only B." or "B has started." if we merge them into one function? (basically in current logic, the function itself stands for the action coming from whom)

Reply:
Yeah, after rethinking, we could combine them into one method with input parameter, some similar logic to:

```
showNotification(fromRecordingTrigger:Bool) {
...
if (isRecordingActive) {
            if (isTranscriptionActive) {
               ...
            } else {
                fromRecordingTrigger ? setRecordingActivityNotification(getText(R.string.start_recording)) : setRecordingActivityNotification(getText(R.string.stop_transcribing_while_recording));
.....
```

Reply:
Perhaps something like this:    

    private enum BannerType {
        RECORDING,
        TRANSCRIPTION
    }

    private void showBanner(BannerType bannerType) {
        if (callingContext.isRecordingActive && callingContext.isTranscriptionActive) {
            setBannerText(getText(R.string.start_recording_and_transcribing))
            return;
        }

        if (bannerType == BannerType.RECORDING) {
            if (callingContext.isRecordingActive) {
                setBannerText(getText(R.string.start_recording))
            } else {
                callingContext.isTranscriptionActive
                        ? setBannerText(getText(R.string.stop_recording_while_transcribing))
                        : setBannerText(getText(R.string.stop_recording));
            }
        } else if (bannerType == BannerType.TRANSCRIPTION) {
            // Similar logic for transcription
        }
    }

Reply:
Yea, addressed. Originally I thought the code in this way doesn't save lines of code. but yea, it saves number of functions and merged some cases.